jps-plugin/src/org/intellij/erlang/jps/builder/ErlangPrepareDependenciesTarget.java

```diff
@@ -0,0 +1,127 @@
+
+  private final JpsProject myProject;
+
+  private ErlangPrepareDependenciesTarget(@NotNull JpsProject project,@NotNull ErlangPrepareDependenciesTargetType targetType) {
+    super(targetType);
+    myProject = project;
+  }
+
+  public static ErlangPrepareDependenciesTarget getInstance(@NotNull JpsProject project,
+                                                            @NotNull ErlangPrepareDependenciesTargetType targetType) {
+    if (ourInstance == null) {
+      ourInstance = new ErlangPrepareDependenciesTarget(project, targetType);
+    }
+    return ourInstance;
+  }
+
+  @Override
+  public String getId() {
+    return myProject.getName();
+  }
+
+  @Override
+  public Collection<BuildTarget<?>> computeDependencies(BuildTargetRegistry targetRegistry,
+                                                        TargetOutputIndex outputIndex) {
+    return Collections.emptyList();
+  }
+
+  @NotNull
+  @Override
+  public List<ErlangSourceRootDescriptor> computeRootDescriptors(@NotNull JpsModel model,
+                                                                 ModuleExcludeIndex index,
+                                                                 IgnoredFileIndex ignoredFileIndex,
+                                                                 @NotNull BuildDataPaths dataPaths) {
+    List<ErlangSourceRootDescriptor> result = new ArrayList<ErlangSourceRootDescriptor>();
+    for (JpsModule module : model.getProject().getModules()) {
+      for (JpsTypedModuleSourceRoot<JavaSourceRootProperties> root : module.getSourceRoots(JavaSourceRootType.SOURCE)) {
+        result.add(new ErlangSourceRootDescriptor(root.getFile(), this, false));
+      }
+      for (JpsTypedModuleSourceRoot<JavaSourceRootProperties> root : module.getSourceRoots(JavaSourceRootType.TEST_SOURCE)) {
+        result.add(new ErlangSourceRootDescriptor(root.getFile(), this, true));
+      }
+      for (JpsTypedModuleSourceRoot<JpsDummyElement> root : module.getSourceRoots(ErlangIncludeSourceRootType.INSTANCE)) {
+        result.add(new ErlangSourceRootDescriptor(root.getFile(), this, false));
+      }
+    }
+    return result;
+  }
+
+  @Override
+  public int hashCode() {
+    return super.hashCode();
+  }
+
+  @Nullable
+  @Override
+  public ErlangSourceRootDescriptor findRootDescriptor(String rootId, BuildRootIndex rootIndex) {
+    return ContainerUtil.getFirstItem(rootIndex.getRootDescriptors(new File(rootId), Collections.singletonList(getErlangTargetType()), null));
+  }
+
+  @NotNull
+  @Override
+  public String getPresentableName() {
+    return NAME;
+  }
+
+  @NotNull
+  @Override
+  public Collection<File> getOutputRoots(@NotNull CompileContext context) {
+    return Collections.emptyList();
+  }
+
+  @NotNull
+  public JpsProject getProject() {
+    return myProject;
+  }
+
+  private ErlangPrepareDependenciesTargetType getErlangTargetType() {
+    return (ErlangPrepareDependenciesTargetType) getTargetType();
+  }
```

Comment:
`getErlangTargetType` is not static method and it use information about type, I can extract utility method, but it will be look weird like this
`public <P extends BuildTarget<ErlangSourceRootDescriptor>,T extends BuildTargetType<P>> T getSpecifiedTargetType(P buildTarget){`
    `return (T) buildTarget.getTargetType();`
 ` }`